06 09 2023  as delete app registration

AcceleryntSecurityDev opened the pull request:
Change(s):
   - New playbook

   Reason for Change(s):
   - New playbook

   Testing Completed:
   - Yes


<img width="1004" alt="image" src="https://github.com/Azure/Azure-Sentinel/assets/101294083/2c8c7677-039c-43b7-bfb3-b1552dee33a1">

rahul0216 reviewed (commented):
Playbook looks good. Couple of suggestions,
1. I think In the readme file, this need to be highlighted as warning that all the app registration with the same name may get deleted.
2. Can you also add comment to incident via logic app, so that SOC engineer get to know that action is completed and these app registration has been deleted?

AcceleryntSecurityDev commented:
@rahul0216 I have implemented your suggestions, noting that adding comments to incidents requires the playbook to have the microsoft sentinel contributor role, so I have added documentation for this as well.

rahul0216 commented on Playbooks/AS-Delete-App-Registration/README.md:
```diff
@@ -0,0 +1,176 @@
+This playbook is intended to be run from a Microsoft Sentinel incident. If any app registration entities are found (i.e. any entities where kind == CloudApplication), they will be deleted. This playbook matches by name, since a unique app registration ID cannot currently be pulled into the entity list, so if there are multiple app registrations exactly matching the name(s) of the CloudApplication ...
+
+> **Warning**
+> If there are multiple app registrations exactly matching the name(s) of the CloudApplication entities in the Microsoft icident used to run this playbook, all of those app registrations will be deleted
```
Correct the spelling mistake 'icident'

rahul0216 reviewed (commented):
Changes look good. There is one spelling mistake in readme files. Please correct that.

rahul0216 reviewed (commented):
Changes look good to merge.
